File: tourcount/src/main/java/com/wmstein/tourcount/widgets/EditHeadWidget.java
package com.wmstein.tourcount.widgets;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.wmstein.tourcount.R;

import java.util.Objects;

/***************************************************
 * EditHeadWidget.java used by EditMetaActivity.java
 * Created by wmstein for com.wmstein.tourcount on 2016-03-31,
 * last edited on 2019-02-12
 */
public class EditHeadWidget extends LinearLayout
{
    private final TextView widget_co1; // used for country title
    private final EditText widget_co2; // used for country
    private final TextView widget_name1; // used for observer title
    private final EditText widget_name2; // used for observer

    public EditHeadWidget(Context context, AttributeSet attrs)
    {
        super(context, attrs);
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        Objects.requireNonNull(inflater).inflate(R.layout.widget_edit_head, this, true);
        widget_co1 = findViewById(R.id.widgetCo1);
        widget_co2 = findViewById(R.id.widgetCo2);
        widget_name1 = findViewById(R.id.widgetName1);
        widget_name2 = findViewById(R.id.widgetName2);
    }

    // Following the SETS
    // country
    public void setWidgetCo1(String title)
    {
        widget_co1.setText(title);
    }

    public void setWidgetCo2(String name)
    {
        widget_co2.setText(name);
    }

    // name
    public void setWidgetName1(String title)
    {
        widget_name1.setText(title);
    }

    public void setWidgetName2(String name)
    {
        widget_name2.setText(name);
    }

    // Following the GETS
    // country
    public String setWidgetCo2()
    {
        return widget_co2.getText().toString();
    }

    // name
    public String setWidgetName2()
    {
        return widget_name2.getText().toString();
    }

}

File: tourcount/src/main/java/com/wmstein/tourcount/widgets/EditHeadWidget.kt
package com.wmstein.tourcount.widgets

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.TextView
import com.wmstein.tourcount.R
import java.util.Objects

/**********************************
 * EditHeadWidget used by CountOptionsActivity
 * Created by wmstein on 2016-02-18,
 * last edited in Java on 2020-09-19,
 * converted to Kotlin on 2024-05-11,
 * last edited on 2024-11-25
 */
class EditHeadWidget(context: Context, attrs: AttributeSet?) : LinearLayout(context, attrs) {
    private val sp_list_title: TextView
    private val sp_list_name: EditText
    private val notes_title: TextView
    private val notes_name: EditText

    val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

    init {
        Objects.requireNonNull(inflater).inflate(R.layout.widget_edit_head, this, true)
        sp_list_title = findViewById(R.id.spListTitle)
        sp_list_name = findViewById(R.id.spListName)
        notes_title = findViewById(R.id.notesTitle)
        notes_name = findViewById(R.id.notesName)
    }

    fun setSpListTitle(title: String?) {
        sp_list_title.text = title
    }

    var spListName: String?
        get() = sp_list_name.text.toString()
        set(name) {
            sp_list_name.setText(name)
        }

    fun setNotesTitle(title: String?) {
        notes_title.text = title
    }

    var notesName: String?
        get() = notes_name.text.toString()
        set(name) {
            notes_name.setText(name)
        }

}
